src/JMS/Serializer/Metadata/Driver/AnnotationDriver.php

```diff
@@ -160,10 +160,12 @@ public function loadMetadataForClass(\ReflectionClass $class)
                         $isExpose = true;
                         if (null !== $annot->if) {
                             $propertyMetadata->excludeIf = "!(" . $annot->if . ")";
+                            $classMetadata->usingExpression = true;
```

Comment:
You can check this in addPropertyMetadata of the ClassMetadata class, no need to have this in each driver then.